MapboxNavigation/NavigationViewController.swift

```diff
@@ -154,6 +154,19 @@ public protocol NavigationViewControllerDelegate {
      Returns the center point of the user course view in screen coordinates relative to the map view.
      */
     @objc optional func navigationViewController(_ navigationViewController: NavigationViewController, mapViewUserAnchorPoint mapView: NavigationMapView) -> CGPoint
+    
+    /**
+     Returns whether the `BottomBannerView` in the `RouteMapViewController` should show a cancel button.
+     
+     If implemented, this method is called at `RouteMapViewController`'s `viewWillAppear()` time. This is used to initally determine whether the cancel button should be shown when the view first appears. If you need to set the visibility of the cancel button after this point, manipulate the `BottomBannerView.isShowingCancelButton` property directly.
+     
+     - parameter navigationViewController: The navigation view controller that needs to know wheather or not to show the cancel button.
+     - parameter bottomBanner: The bottom banner containing the cancel button.
+     - note: If this method is not implemented, the default behavior is to show the cancel button if the `NavigationViewController` was modally presented.
+     - returns: True to show the cancel button, False to hide.
+     */
+    @objc optional func navigationViewController(_ navigationViewController: NavigationViewController, shouldShowCancelButtonIn bottomBanner: BottomBannerView) -> Bool
```

Comment:
This method bridges to Objective-C as `-navigationViewController:shouldShowCancelButtonIn:`. To conform to Objective-C naming conventions, we’ll need `@objc(navigationViewController:shouldShowCancelButtonInBottomBannerView:)`.

However, I’m not sure about passing the BottomBannerView as a parameter. Can you think of a situation in which the implementation would need to consult a property on BottomBannerView to decide whether to return true or false? If that does indeed turn out to be useful, we should make the BottomBannerView accessible via NavigationViewController, ideally by making `NavigationViewController.routeMapViewController` public. That would be useful in its own right.